spec/index.bs

```diff
@@ -2036,12 +2043,150 @@ dictionary GPUShaderModuleDescriptor : GPUObjectDescriptorBase {
+
+            1. Let |group| be |resource|'s "group" decoration.
+            1. Let |binding| be |resource's "binding" decoration.
+            1. Let |entry| be a new {{GPUBindGroupLayoutEntry}}.
+            1. Set |entry|.{{GPUBindGroupLayoutEntry/binding}} to |binding|.
+            1. Set |entry|.{{GPUBindGroupLayoutEntry/visibility}} to |shaderStage|.
+            1. If |resource| is for a sampler binding:
+
+                    1. Set |entry|.{{GPUBindGroupLayoutEntry/type}} to
+                        {{GPUBindingType/sampler}}.
+
+            1. If |resource| is for a comparison sampler binding:
+
+                1. Set |entry|.{{GPUBindGroupLayoutEntry/type}} to {{GPUBindingType/comparison-sampler}}.
+
+            1. If |resource| is for a buffer binding:
+
+                1. Set |entry|.{{GPUBindGroupLayoutEntry/hasDynamicOffset}} to false.
+
+                1. If |resource| is for a uniform buffer:
+
+                    1. Set |entry|.{{GPUBindGroupLayoutEntry/type}} to {{GPUBindingType/uniform-buffer}}.
+
+                1. If |resource| is for a read-only storage buffer:
+
+                    1. Set |entry|.{{GPUBindGroupLayoutEntry/type}} to {{GPUBindingType/readonly-storage-buffer}}.
+
+                1. If |resource| is for a storage buffer:
+
+                    1. Set |entry|.{{GPUBindGroupLayoutEntry/type}} to {{GPUBindingType/storage-buffer}}.
+
+            1. If |resource| is for a texture binding:
+
+                1. Set |entry|.{{GPUBindGroupLayoutEntry/textureComponentType}} to |resource|'s component type.
+                1. Set |entry|.{{GPUBindGroupLayoutEntry/viewDimension}} to |resource|'s dimension.
+                1. If |resource| is multisampled:
+
+                    1. Set |entry|.{{GPUBindGroupLayoutEntry/multisampled}} to true.
+
+                1. If |resource| is for a sampled texture:
+
+                    1. Set |entry|.{{GPUBindGroupLayoutEntry/type}} to {{GPUBindingType/sampled-texture}}.
+
+                1. If |resource| is for a read-only storage texture:
+
+                    1. Set |entry|.{{GPUBindGroupLayoutEntry/type}} to {{GPUBindingType/readonly-storage-texture}}.
+                    1. Set |entry|.{{GPUBindGroupLayoutEntry/storageTextureFormat}} to |resource|'s format.
+
+                1. If |resource| is for a write-only storage texture:
+
+                    1. Set |entry|.{{GPUBindGroupLayoutEntry/type}} to {{GPUBindingType/writeonly-storage-texture}}.
+                    1. Set |entry|.{{GPUBindGroupLayoutEntry/storageTextureFormat}} to |resource|'s format.
+
+            1. If |groupDescs|[|group|] has an entry |previousEntry| with {{GPUBindGroupLayoutEntry/binding}} equal to |binding|:
+
+                1. If |previousEntry| is equal to |entry| up to {{GPUBindGroupLayoutEntry/visibility}}:
+
+                    1. Or |entry|.{{GPUBindGroupLayoutEntry/visibility}} into |previousEntry|.{{GPUBindGroupLayoutEntry/visibility}}
```

Comment:
should be reworded, `Or` as a binary operation is not clear here. Perhaps, "add a bitflag of ... to the ... mask" ?

Reply:
Reworded this.